misc/dbt-materialize/dbt/include/materialize/macros/adapters.sql

```diff
@@ -84,6 +84,7 @@
     from mz_objects o
     join mz_schemas s on o.schema_id = s.id and s.name = '{{ schema_relation.schema }}'
     join mz_databases d on s.database_id = d.id and d.name = '{{ schema_relation.database }}'
+    where type in (select type from mz_relations union select 'index' union select 'sink')
```

Comment:
This could explicitly list the values instead of using a subquery, WDYT?

Reply:
Yeah I liked the idea of being clear that it's relations + index + sink, but I also don't like doing a subquery.
Querying relations and unioning with index and sink seems even worse.

I'm flexible here on listing the values and adding a comment to explain. Do we plan on ever adding more relations? My concern would be that it wouldn't get automatically propagated with the explicit list.

(Note: I realized this would be `distinct(type)` actually if we keep the subquery).

Reply:
> Do we plan on ever adding more relations?

That's unlikely, and would require us to add code for _e.g_. new custom materializations! Having the values listed out also makes it more straightforward to understand what to change where, in such cases.